### ISSUE TITLE
TouchBar scripts running on button hold

### ISSUE THREAD

**Issue by Sam1066** [fifafu/BetterTouchTool#2151]
**IMPORTANT:** Don't post private requests here (e.g. license issues, etc.). Please send me an email at boastr.net@gmail.com instead.

---

#### Description of bug/feature request/question:
Upon hold down of a TouchBar button, any associated action is constantly run. For example, I have a button that runs a bash script. When that button is pressed, the time that my finger is on the Touchbar is sufficient time for the script to run twice. If I hold the button down for a few seconds, the script then runs countless times.

#### Device information:

Type of Mac:
macOS version: 
BetterTouchTool version: 

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

---

**Note:** Before bug reporting, please make sure you have the latest version of BetterTouchTool and that you have already tried to restart your system :-). If you encounter a crash, please attach a crash log from the macOS Console.app from the "User Diagnostic Reports" section.

**Comment by fifafu** [comment]
Yes, currently it just behaves like a normal keyboard key which would also repeat when being held down. However it should be easy for me to add an option to disable this functionality.

**Comment by Sam1066** [comment]
Awesome! Thank you! I would appreciate that a ton

- Sam

On Monday, February 26, 2018, 9:36 pm, Andreas Hegenberg <notifications@github.com> wrote:


Yes, currently it just behaves like a normal keyboard key which would also repeat when being held down. However it should be easy for me to add an option to disable this functionality.

**Comment by cameroncooke** [comment]
@fifafu Any update on this as I have the same issue, as it can do disastrous thing to your system. Need a way to turn off key repeat on touch bar actions.

**Comment by fifafu** [comment]
v2.430 alpha now has options to set repeatRate, repeatDelay and also the option to set a different action for "long presses"
